assert/rich_assert.go

```diff
@@ -70,7 +74,19 @@ func newOperands[T Number](left, right T) any {
 	case float32, float64:
 		return numericOperands[float64]{float64(left), float64(right)}
 	}
-	return nil
+
+	// Got this far, and still not sure?
+	leftVal := reflect.ValueOf(left)
+	rightVal := reflect.ValueOf(right)
+	if leftVal.CanUint() {
+		return numericOperands[uint64]{leftVal.Uint(), rightVal.Uint()}
+	} else if leftVal.CanInt() {
+		return numericOperands[int64]{leftVal.Int(), rightVal.Int()}
+	} else if leftVal.CanFloat() {
+		return numericOperands[float64]{leftVal.Float(), rightVal.Float()}
+	} else {
+		return nil
```

Comment:
This last case should be unreachable, right? Maybe print a warning message or even panic here?

Reply:
Yes I'll print a warning, and return nil.  I don't really want to panic when an assertion is not able to be forwarded

Reply:
Also, added a commit to ensure that when instrumentation is run with the `-assert_only` option, the specified source files are included for cataloging.